src/NServiceBus.Core.Tests/Sagas/ComponentTests/SagaWithCorrelationPropertyData.cs

```diff
@@ -0,0 +1,29 @@
+namespace NServiceBus.Persistence.ComponentTests
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class SagaWithCorrelationProperty : Saga<SagaWithCorrelationPropertyData>, IAmStartedByMessages<SagaCorrelationPropertyStartingMessage>
+    {
+        public Task Handle(SagaCorrelationPropertyStartingMessage message, IMessageHandlerContext context)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override void ConfigureHowToFindSaga(SagaPropertyMapper<SagaWithCorrelationPropertyData> mapper)
+        {
+            mapper.ConfigureMapping<SagaCorrelationPropertyStartingMessage>(m => m.CorrelatedProperty).ToSaga(s => s.CorrelatedProperty);
+        }
+    }
+    public class SagaWithCorrelationPropertyData : ContainSagaData
+    {
+        public string CorrelatedProperty { get; set; }
+
+        public DateTime DateTimeProperty { get; set; }
```

Comment:
Perhaps just a string to make this less prone to screw up things?

Or should we try to establish a base line for what persisters should always support?